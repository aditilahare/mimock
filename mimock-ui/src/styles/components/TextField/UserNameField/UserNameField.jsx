import React from 'react';
import PropTypes from 'prop-types';
import { UserNameFieldContainer } from './UserNameField.style';

function UserNameField({
	dataTestid,
	placeholder,
	autoComplete,
	value,
	onChangeHandler,
	onBlurHandler,
	onFocusHandler,
}) {
	return (
		<UserNameFieldContainer
			data-testid={dataTestid}
			placeholder={placeholder}
			autoComplete={autoComplete}
			value={value}
			onChange={onChangeHandler}
			onBlur={onBlurHandler}
			onFocus={onFocusHandler}
		/>
	);
}

UserNameField.propTypes = {
	dataTestid: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired,
	onChangeHandler: PropTypes.func,
	onBlurHandler: PropTypes.func,
	onFocusHandler: PropTypes.func,
	placeholder: PropTypes.string,
	autoComplete: PropTypes.bool,
};

export default UserNameField;
