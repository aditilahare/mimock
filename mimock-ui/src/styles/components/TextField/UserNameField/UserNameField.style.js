import tw from 'tailwind-styled-components';

export const UserNameFieldContainer = tw.input`
    w-full
    my-2
    p-2
    outline-none
    rounded-md
    border-2
    border-gray-300
`;
