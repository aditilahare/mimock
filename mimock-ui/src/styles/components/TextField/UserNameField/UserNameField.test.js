import React from 'react';
import { render } from '@testing-library/react';
import UserNameField from './UserNameField';

describe('Text Field', () => {
	let tree;

	it('should render basic text field', async () => {
		tree = await render(
			<UserNameField
				dataTestid='login-username-input'
				placeholder='username'
				required
				autoComplete='off'
				value='username'
				onChangeHandler={() => {
					console.log('Change handler');
				}}
				onBlurHandler={() => {
					console.log('Blur handler');
				}}
				onFocusHandler={() => {
					console.log('Focus handler');
				}}
			/>
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('login-username-input')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
