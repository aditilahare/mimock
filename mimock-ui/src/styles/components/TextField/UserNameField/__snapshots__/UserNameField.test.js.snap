// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Field should render basic text field 1`] = `
<div>
  <input
    autocomplete="off"
    class="w-full my-2 p-2 outline-none rounded-md border-2 border-gray-300"
    data-testid="login-username-input"
    placeholder="username"
    value="username"
  />
</div>
`;
